test/logging/logging.go

```diff
@@ -113,29 +119,49 @@ func initializeLogger(logVerbose bool) {
 // using the provided context as a name. This will also register the logger as a metric exporter,
 // which is unfortunately global, so calling `GetContextLogger` will have the side effect of
 // changing the context in which all metrics are logged from that point forward.
-func GetContextLogger(context string) *zap.SugaredLogger {
+func GetContextLogger(context string) *BaseLogger {
 	// If there was a previously registered exporter, unregister it so we only emit
 	// the metrics in the current context.
 	if exporter != nil {
 		view.UnregisterExporter(exporter)
 		trace.UnregisterExporter(exporter)
 	}
 
-	logger := baseLogger.Named(context)
+	logger := baseLogger.Logger.Named(context)
 
 	exporter = &zapMetricExporter{logger: logger}
 	view.RegisterExporter(exporter)
 	trace.RegisterExporter(exporter)
 
-	return logger
+	return &BaseLogger{Logger: logger}
 }
 
-// LogResourceObject logs the resource object with the resource name and value
-func LogResourceObject(logger *zap.SugaredLogger, value ResourceObjects) {
```

Comment:
why move it out to a separate util.go file?
If we don't plan to move more functions to util.go then better to keep it within logging.go. Arguably, it may even stir questions about what should be in util vs. what not, and will cause confusion if it just contains the log function.

Reply:
This function is specific to serving while the logging library will be common to all repos in knative. I needed some common place in knative/serving/test to put this and probably other helper methods so created this file.